functions/lib/emails/cancel_event.py

```
@@ -0,0 +1,72 @@
+        # Retry once more on rate limit after waiting 1 second
+        if response.status_code == 429:
+            time.sleep(1)
+            response = requests.post(
+                "https://app.loops.so/api/v1/transactional",
+                data=json.dumps(body),
+                headers=headers,
+            )
+
+        if response.status_code != 200:
+            logger.error(
+                f"Failed to send cancellation email for orderId={order_id}, body={response.text}"
```

Ruff (0.14.10)

Logging statement uses f-string

(G004)